packages/cli/src/services/community.service.ts

```diff
@@ -0,0 +1,306 @@
+				if (errorMessage.includes(npmMessage)) throw new Error(n8nMessage);
+			});
+
+			Logger.warn('npm command failed', { errorMessage });
+
+			throw new Error(PACKAGE_FAILED_TO_INSTALL);
+		}
+	}
+
+	matchPackagesWithUpdates(
+		packages: InstalledPackages[],
+		updates?: CommunityPackages.AvailableUpdates,
+	) {
+		if (!updates) return packages;
+
+		return packages.reduce<PublicInstalledPackage[]>((acc, cur) => {
+			const publicPackage: PublicInstalledPackage = { ...cur };
+
+			const update = updates[cur.packageName];
+
+			if (update) publicPackage.updateAvailable = update.latest;
+
+			acc.push(publicPackage);
+
+			return acc;
+		}, []);
+	}
+
+	matchMissingPackages(installedPackages: PublicInstalledPackage[], missingPackages: string) {
+		const missingPackagesList = missingPackages
+			.split(' ')
+			.map((name) => {
+				try {
+					// Strip away versions but maintain scope and package name
+					const parsedPackageData = this.parseNpmPackageName(name);
+					return parsedPackageData.packageName;
+				} catch {
+					return;
+				}
+			})
+			.filter((i): i is string => i !== undefined);
+
+		const hydratedPackageList: PublicInstalledPackage[] = [];
+
+		installedPackages.forEach((installedPackage) => {
+			const hydratedInstalledPackage = { ...installedPackage };
+
+			if (missingPackagesList.includes(hydratedInstalledPackage.packageName)) {
+				hydratedInstalledPackage.failedLoading = true;
+			}
+
+			hydratedPackageList.push(hydratedInstalledPackage);
+		});
+
+		return hydratedPackageList;
+	}
+
+	async checkNpmPackageStatus(packageName: string) {
+		const N8N_BACKEND_SERVICE_URL = 'https://api.n8n.io/api/package';
+
+		try {
+			const response = await axios.post<CommunityPackages.PackageStatusCheck>(
+				N8N_BACKEND_SERVICE_URL,
+				{ name: packageName },
+				{ method: 'POST' },
+			);
+
+			if (response.data.status !== NPM_PACKAGE_STATUS_GOOD) return response.data;
+		} catch {
+			// service unreachable, do nothing
+		}
+
+		return { status: NPM_PACKAGE_STATUS_GOOD };
+	}
+
+	hasPackageLoaded(packageName: string) {
+		const missingPackages = config.get('nodes.packagesMissing') as string | undefined;
+
+		if (!missingPackages) return true;
+
+		return !missingPackages
+			.split(' ')
+			.some(
+				(packageNameAndVersion) =>
+					packageNameAndVersion.startsWith(packageName) &&
+					packageNameAndVersion.replace(packageName, '').startsWith('@'),
+			);
+	}
+
+	removePackageFromMissingList(packageName: string) {
+		try {
+			const failedPackages = config.get('nodes.packagesMissing').split(' ');
+
+			const packageFailedToLoad = failedPackages.filter(
+				(packageNameAndVersion) =>
+					!packageNameAndVersion.startsWith(packageName) ||
+					!packageNameAndVersion.replace(packageName, '').startsWith('@'),
+			);
+
+			config.set('nodes.packagesMissing', packageFailedToLoad.join(' '));
+		} catch {
+			// do nothing
+		}
+	}
+
+	async setMissingPackages(
+		loadNodesAndCredentials: LoadNodesAndCredentials,
```

Comment:
`CommunityService` depends on `LoadNodesAndCredentials` and vice versa. Passing as arg is only a workaround. How best to address this?

Reply:
Discussed with Omar - there is no other way apparently.